internal/service/adapters.go

```diff
@@ -0,0 +1,147 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-pg/pg"
+	legacyredis "github.com/go-redis/redis"
+	"github.com/go-redis/redis/v8"
+	clocktime "github.com/topfreegames/maestro/internal/adapters/clock/time"
+	instancestorageredis "github.com/topfreegames/maestro/internal/adapters/instance_storage/redis"
+	operationstorageredis "github.com/topfreegames/maestro/internal/adapters/operation_storage/redis"
+	roomstorageredis "github.com/topfreegames/maestro/internal/adapters/room_storage/redis"
+	kubernetesruntime "github.com/topfreegames/maestro/internal/adapters/runtime/kubernetes"
+	portallocatorrandom "github.com/topfreegames/maestro/internal/adapters/port_allocator/random"
+	schedulerstoragepg "github.com/topfreegames/maestro/internal/adapters/scheduler_storage/pg"
+	"github.com/topfreegames/maestro/internal/config"
+	"github.com/topfreegames/maestro/internal/core/ports"
+	"github.com/topfreegames/maestro/internal/core/entities"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// configurations paths for the adapters
+const (
+	// Kubernetes runtime
+	runtimeKubernetesMasterUrlPath  = "adapters.runtime.kubernetes.masterUrl"
+	runtimeKubernetesKubeconfigPath = "adapters.runtime.kubernetes.kubeconfig"
+	// Redis operation storage
+	operationStorageRedisUrlPath = "adapters.operationStorage.redis.url"
+	// Redis room storage
+	roomStorageRedisUrlPath = "adapters.roomStorage.redis.url"
+	// Redis instance storage
+	instanceStorageRedisUrlPath = "adapters.intanceStorage.redis.url"
+	instanceStorageRedisScanSizePath = "adapters.intanceStorage.redis.scanSize"
+	// Random port allocator
+	portAllocatorRandomRangePath = "adapters.portallocator.random.range"
+	// Postgres scheduler storage
+	schedulerStoragePostgresUrlPath = "adapters.schedulerStorage.postgres.url"
+)
+
+func NewRuntimeKubernetes(ctx context.Context, c config.Config) (ports.Runtime, error) {
+	clientset, err := createKubernetesClient(
+		ctx,
+		c.GetString(runtimeKubernetesMasterUrlPath),
+		c.GetString(runtimeKubernetesKubeconfigPath),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize Kubernetes runtime: %w", err)
+	}
+
+	return kubernetesruntime.New(clientset), nil
+}
+
+func NewOperationStorageRedis(ctx context.Context, clock ports.Clock, c config.Config) (ports.OperationStorage, error) {
+	client, err := createRedisClient(ctx, c.GetString(operationStorageRedisUrlPath))
```

Comment:
Since we're treating every adapter as different solutions (which may use different Redis instances), I'd say that if we want to have only a single Redis client, the best way is to have only a single configuration. Still, I'm not sure if it fits good the structure of the adapters. WDYT?

Reply:
Different redis instances could be treated with different redis clients (globally) my only concern here is the connection time it will consume in order to process each redis operation. What could take more than we expect.

Reply:
So, imagine we have in a game:

- 3 schedulers, we're going to open (async) a lot of connections (redis is limited about it) in order to process operations that could be executed with a single client (or a centralized pool).